source/distributed/solution_transfer.cc

```diff
@@ -137,6 +137,12 @@ namespace parallel
       prepare_for_coarsening_and_refinement(
         const std::vector<const VectorType *> &all_in)
     {
+#  ifdef DEBUG
+      for (unsigned int i = 0; i < all_in.size(); ++i)
+        Assert(all_in[i]->size() == dof_handler->n_dofs(),
```

Comment:
I think you can remove the `#ifdef`. The `Assert` expands to nothing in release mode and the compiler can hopefully see that the `all_in.size()` call doesn't change any state of `all_in` so that the whole loop becomes a no-op.

Reply:
I don't have a preference on the `DEBUG` macros here. I like being more explicit about sections that only happen in `DEBUG` mode, but right here and in the beginning of the function it should be obvious. Will comply and remove them!